api-extractor/src/ResolvedApiItem.ts

```diff
@@ -0,0 +1,107 @@
+import ApiItem, { ApiItemKind } from './definitions/ApiItem';
+import ApiItemContainer from './definitions/ApiItemContainer';
+import { ApiTag } from './definitions/ApiDocumentation';
+import { IDocElement, IParam } from './IDocElement';
+import { IDocItem } from './IDocItem';
+import ApiJsonFile from './generators/ApiJsonFile';
+
+/**
+ * A class to abstract away the difference between an item from our public API that could be 
+ * represented by either an ApiItem or an IDocItem that is retrieved from a JSON file.
+ */
+export default class ResolvedApiItem {
+  public kind: ApiItemKind;
+  public summary: IDocElement[];
+  public remarks: IDocElement[];
+  public deprecatedMessage: IDocElement[];
+  public isBeta: boolean;
+  public params: {[name: string]: IParam};
+  public returnsMessage: IDocElement[];
+  public members: { [name: string]: ApiItem | IDocItem};
```

Comment:
You're right, it's not neccessary. I added the members property to deal with the question you raised in the comment above this one.